src/page_table.rs

```diff
@@ -18,16 +20,15 @@ pub(crate) fn map_kernel(
         map_segment(segment, kernel_start, page_table, frame_allocator)?;
     }
 
-    // create a stack
-    // TODO create a stack range dynamically (based on where the kernel is loaded)
-    let stack_start = Page::containing_address(VirtAddr::new(0x57AC_0000_0000));
+    // Create a stack
     let stack_size: u64 = 512; // in pages
     let stack_end = stack_start + stack_size;
 
     let flags = PageTableFlags::PRESENT | PageTableFlags::WRITABLE;
     let region_type = MemoryRegionType::KernelStack;
 
-    for page in Page::range(stack_start, stack_end) {
+    // Leave the first page unmapped as a 'guard page'
+    for page in Page::range(stack_start + 1, stack_end) {
```

Comment:
Nit: This makes the actual stack size one page smaller than `stack_size`. I don't think that it matters much, but how about doing a `let stack_start = stack_start + 1` before adding the `stack_size`?

Reply:
I had it like that so that it wouldn’t spill over into a new p1 table in the case that the boot info is mapped to the page right before the stack.

Unrelated: In general 512 pages for the stack seems like quite a lot; for reference the Linux kernel only uses 2 pages per kernel stack.

Reply:
I'm happy to change that number. Ideally, this would be customizable through an environment variable too, so that we can default to a small stack and let people increase the size if needed.

Reply:
I'll add that to the roadmap